#DPATCHLEVEL=0
# Fix timeout when additional messages are displayed with negative duration.
# xosd_display(SOMETHING)
# 	starts the timer, osd->timeout_start gets set
# xosd_set_timeout(-1)
# 	just sets osd->timeout to -1
# xosd_display(SOMETHING)
# 	signals the event thread and blocks until SHOWN (!)
# event_loop()
# 	updates the display
# 	sets osd->mapped := 1 (*)
# 	DOESN'T clear the old osd->timeout_start, so it's still set
# 	Calculates tv as (now-(osd->timeout)) = (now + 1) (future!)
# 	Checks if tv is after osd->timeout_start: YES
# 	Since osd->mapped, clear the timer an osd->mapped := 0
# 	Jump back to start
# Patch by Nicolas Boullis <nboullis@debian.org>, closes: #277027
--- src/libxosd/xosd.c
+++ src/libxosd/xosd.c
@@ -380,6 +380,8 @@
       osd->update = UPD_none;
       if (osd->timeout > 0)
         gettimeofday(&osd->timeout_start, NULL);
+      else
+        timerclear(&osd->timeout_start);
     } else if (timerisset(&osd->timeout_start)) {
       gettimeofday(&tv, NULL);
       tv.tv_sec -= osd->timeout;
